functions.go

```diff
@@ -137,8 +137,8 @@ func init() {
 }
 
 // Factorial calculates the factorial of number n
-func Factorial(n float64) float64 {
-	if n == 0 {
+func Factorial(n int64) int64 {
+	if n <= 1 {
```

Comment:
On this way you avoid a further recursion.
The Value 1 will cause a recursion which will also return 1. So you multiply by 1 two times.
A second goal is the avoiding of negative values.